lib/rules/number-max-precision/index.js

```diff
@@ -72,47 +85,122 @@ const rule = (primary, secondaryOptions) => {
 				return;
 			}
 
-			valueParser(value).walk((valueNode) => {
-				const { unit } = getDimension(valueNode);
+			parseListOfComponentValues(tokenize({ css: value })).forEach((componentValue) => {
+				/** @type {{ ignored: boolean, precision: number }} */
```

Comment:
[suggest] When specifying a primary option type like this:

```diff
-/** @type {import('stylelint').Rule} */
+/** @type {import('stylelint').Rule<number>} */
```

This `@type` annotation seems no longer needed:

```suggestion
```